### ISSUE TITLE
DataGrid Pagination not screenreader (VoiceOver) accessible

### ISSUE THREAD

**Issue by sheamus-hei** [mui/mui-x#6383]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

At the end of the data table, intuitively VoiceOver should read the text in the Pagination row ("Rows per page" etc), but instead it skips that row entirely and just says "End of Table." The select and buttons are accessible through the list of form controls, but this is not ideal. 

### Expected behavior 🤔

Pagination feature should be read by VoiceOver both when traversing through the table and from the form control list.

### Steps to reproduce 🕹

Link to live example:
https://mui.com/x/react-data-grid/pagination/

Steps:
1. Navigate to the page and activate VoiceOver by pressing `cmd` + `F5`
2. Sift through the examples until you reach the bottom of a table
3. VoiceOver will read the last data cell, but instead of moving focus to the pagination row, it just says "End of table"

It's interesting because the Pagination component on the regular MUI `Table` is perfectly accessible: https://mui.com/x/react-data-grid/pagination/

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by sheamus-hei** [comment]
@m4theushw I think that's correct! The screen reader should read out the pagination contents after the last cell, since that would match the visual experience. It might also be nice to add a "skip to navigation" feature at the top of the table to skip straight to pagination instead of having to scroll through the whole table. As a disclaimer, I am a sighted user and don't rely on screen readers, I only test for basic functionality.
